runtime/src/snapshot_utils.rs

```diff
@@ -1795,6 +1799,19 @@ pub fn process_accounts_package_pre(
     )
 }
 
+/// Filter snapshot storages and retain only the ones with slots _higher than_
+/// `incremental_snapshot_base_slot`.
+pub fn filter_snapshot_storages_for_incremental_snapshot(
+    snapshot_storages: &mut SnapshotStorages,
+    incremental_snapshot_base_slot: Slot,
+) {
+    snapshot_storages.retain(|storage| {
+        storage
+            .first()
+            .map_or(false, |entry| entry.slot() > incremental_snapshot_base_slot)
+    });
```

Comment:
The algorithm is different than the original one in Bank. This is for two reasons:
1. You cannot do a `retain()` within a `retain()` because of an oversight in Rust. Since `retain()` doesn't give mutable access to its iterator, the inner Vec cannot call `retain()`.
2. Since the `SnapshotStorages` is a Vec of Slots, the inner vector will always be for a single slot. So we can more easily filter by just checking the first item.

Reply:
seems reasonable.